src/modules/admin/components/forms/FormTreeView.tsx

```diff
@@ -0,0 +1,267 @@
+  );
+};
+
+interface FormItemDisplay {
+  icon: SvgIconComponent;
+  text: string;
+  backgroundColor: string;
+  textColor: string;
+  hoverColor: string;
+}
+
+interface FormTreeItemProps
+  extends Omit<UseTreeItem2Parameters, 'rootRef'>,
+    Omit<React.HTMLAttributes<HTMLLIElement>, 'onFocus'> {}
+
+const FormTreeItem = React.forwardRef(function FormTreeItem(
+  props: FormTreeItemProps,
+  ref: React.Ref<HTMLLIElement>
+) {
+  const { id, itemId, label, disabled, children, ...other } = props;
+
+  const {
+    getRootProps,
+    getContentProps,
+    getIconContainerProps,
+    getLabelProps,
+    getGroupTransitionProps,
+    status,
+    publicAPI,
+  } = useTreeItem2({ id, itemId, children, label, disabled, rootRef: ref });
+
+  const item = publicAPI.getItem(itemId);
+  const question = getItemDisplayAttrs(item.type);
+
+  // TODO: Add visual tree styling that groups the items together using grey bars.
+  // Should be possible, see example https://mui.com/x/react-tree-view/rich-tree-view/customization/#file-explorer
+
+  // TODO: enable rich text in labels
+
+  return (
+    <TreeItem2Provider itemId={itemId}>
+      <TreeItem2Root {...getRootProps(other)}>
+        <TreeItem2Content
+          {...getContentProps()}
+          sx={{
+            padding: 0,
+            marginBottom: theme.spacing(1),
+            borderStyle: 'solid',
+            borderWidth: 1,
+            borderColor: theme.palette.grey[300],
+          }}
+        >
+          <Box sx={{ flexGrow: 1, display: 'flex', gap: 1 }}>
+            <FormTreeLabel
+              {...getLabelProps({ question })}
+              required={item.required}
+            />
+            <TreeItem2IconContainer
+              {...getIconContainerProps()}
+              sx={{
+                display: 'flex',
+                flexDirection: 'column',
+                mr: 2,
+              }}
+            >
+              <TreeItem2Icon status={status} />
+            </TreeItem2IconContainer>
+          </Box>
+        </TreeItem2Content>
+        {children && (
+          <TreeItem2GroupTransition {...getGroupTransitionProps()} />
+        )}
+      </TreeItem2Root>
+    </TreeItem2Provider>
+  );
+});
+
+interface FormTreeViewProps {
+  definition: FormDefinitionJson;
+}
+const FormTreeView: React.FC<FormTreeViewProps> = ({ definition }) => {
+  // TODO - let's discuss the best way to support nesting
+  // MUI tree view component expects the 'item' prop to be named 'children', and doesn't provide an overrider function.
+  // OS contribution to react mui that provides an override? (but maybe there's a reason they don't allow)
+  // rewriting our DSL to use 'children' instead of 'item'? (but this feels like big lift + bad coupling)
+  // keep and clean up the below adapter, expanding it to n layers of nesting (n = 5?) ?
+  // We might need something like this adapter anyway to consolidate down some functional groups, like SSN
+  const definitionForTree = useMemo(() => {
```

Comment:
This seems okay to me. We can use the existing `modifyFormDefinition` function instead though, something like the below should add `children` to each item recursively.
```
modifyFormDefinition(definition, (item) => (item.children = item.item)
```

We probably _do_ need to place a limit on recursion, though, that should be incorporated somewhere in the form builder. (e.g. at a certain point, you can no longe add a nested group to an existing group)

Reply:
Oh this sounds good. Thanks for pointing it out to me! `modifyFormDefinition` doesn't work out-of-the-box exactly, bc typescript is not happy about appending a new property `children` to `item` since that prop doesn't exist on tyep `FormItem`, but something similar will work -- maybe I will adapt the existing recursive function or write a new one.

I agree about the recursion limit. Added a note about that here so it doesn't get lost: https://github.com/open-path/Green-River/issues/6051